src/Sdk/Cerbos/Sdk/Cloud/V1/HubClientBuilder.cs

```diff
@@ -0,0 +1,109 @@
+
+using System;
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using Grpc.Net.Client;
+using Cerbos.Api.Cloud.V1.ApiKey;
+using Cerbos.Api.Cloud.V1.Store;
+using Cerbos.Sdk.Cloud.V1.Interceptor;
+using Cerbos.Sdk.Cloud.V1.Store;
+using Cerbos.Sdk.Cloud.V1.ApiKey;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Cerbos.Sdk.Cloud.V1
+{
+    public sealed class HubClientBuilder
+    {
+        private string Target { get; }
+        private string CaCertificate { get; set; }
+        private Credentials Credentials { get; set; }
+        private bool Plaintext { get; set; }
+
+        private HubClientBuilder(string target)
+        {
+            Target = target;
+        }
+
+        public static HubClientBuilder ForTarget(string target)
+        {
+            return new HubClientBuilder(target);
+        }
+
+        public HubClientBuilder WithCaCertificate(string path) {
+            CaCertificate = path;
+            return this;
+        }
+
+        public HubClientBuilder WithCredentials(Credentials credentials)
+        {
+            Credentials = credentials;
+            return this;
+        }
+
+        public HubClientBuilder WithCredentials(string clientId, string clientSecret)
+        {
+            Credentials = new Credentials(clientId, clientSecret);
+            return this;
+        }
+
+        public HubClientBuilder WithPlaintext()
+        {
+            Plaintext = true;
+            return this;
+        }
+
+        public IHubClient Build()
+        {
+            if (string.IsNullOrEmpty(Target))
+            {
+                throw new Exception("Target must be specified");
+            }
+
+            if (CaCertificate != null && Plaintext)
+            {
+                throw new Exception("CaCertificate and plaintext must not be specified at the same time");
+            }
+
+            if (Credentials == null)
+            {
+                throw new Exception("Credentials must be specified");
+            }
+
+            var channelOptions = new GrpcChannelOptions{};
+            if (CaCertificate != null)
+            {
+                var handler = new HttpClientHandler();
+                var cert = new X509Certificate(CaCertificate);
+                handler.ClientCertificates.Add(cert);
+                channelOptions.HttpHandler = handler;
+            }
+            else if (!Plaintext)
+            {
+                channelOptions.Credentials = ChannelCredentials.SecureSsl;
+            }
+
+            var authInterceptor = new AuthInterceptor(
+                new ApiKeyClient(
+                    new ApiKeyService.ApiKeyServiceClient(
+                        GrpcChannel.ForAddress(
+                            Target,
+                            channelOptions
+                        )
+                    ),
+                    Resilience.NewInstance().WithCircuitBreaker().WithRetry(StatusCode.ResourceExhausted)
```

Comment:
I don't think calling `Resilience.NewInstance()` is quite right. We actually want a global circuit breaker that's shared across all RPC calls. Otherwise, the circuit breaker won't be able to detect when the client is issuing several _different_ failing requests (e.g. Let's say the client calls `Foo`, `Bar` and `Baz`. If the circuit breaker is not shared, instead of seeing 3 failures, it would only see 1 failure each.)